lib/semantics/check-omp-structure.cc

```diff
@@ -98,6 +98,72 @@ void OmpStructureChecker::Leave(const parser::OpenMPBlockConstruct &) {
   ompContext_.pop_back();
 }
 
+void OmpStructureChecker::Enter(const parser::OpenMPSectionsConstruct &x) {
+  const auto &dir{std::get<parser::Verbatim>(x.t)};
+  OmpContext ct{dir.source};
+  ompContext_.push_back(ct);
+  SetContextDirectiveEnum(OmpDirective::SECTIONS);
```

Comment:
I suggest adding a `PushContext` function that allows you to replace these three lines with:
`PushContext(dir.source, OmpDirective::SECTIONS)`

For the places where you push without setting the directive, it would be just:
`PushContext(dir.source);`

Reply:
Addressed by overloading functions.